packages/flutter_driver/lib/src/driver/driver.dart

```diff
@@ -522,6 +523,54 @@ class FlutterDriver {
     return _getOffset(finder, OffsetType.center, timeout: timeout);
   }
 
+  /// Returns a JSON map of the [DiagnosticsNode] that is associated with the
+  /// [RenderObject] identified by `finder`.
+  ///
+  /// The `subtreeDepth` argument controls how many layers of children will be
+  /// included in the result. It defaults to zero, which means that no children
+  /// of the [RenderObject] identified by `finder` will be part of the result.
+  ///
+  /// The `includeProperties` argument controls whether properties of the
+  /// [DiagnosticsNode]s will be included in the result. It defaults to true.
+  Future<Map<String, Object>> getRenderObjectDiagnostics(
```

Comment:
Could you document the differences between this method and #getWidgetDiagnostics? Like when should use which?

Reply:
Not sure how to answer that question generically. If you're interested in the properties of a Widget use getWidgetDiagnostics, if you want to know the properties of its underlying RenderObject use getRenderObjectDiagnostics.

Reply:
Perhaps something like:

```
[RenderObject]s are responsible for positioning, layout, and painting on the screen, based on the configuration from a [Widget]. Callers that need information about size or position should use this method.

A widget may indirectly create multiple [RenderObject]s, which each implement some aspect of the widget configuration. A 1:1 relationship should not be assumed.
```

And on the widget one:

```
[Widget]s describe configuration for the rendering tree. Individual widgets may create multiple [RenderObject]s to actually layout and paint the desired configuration.  Callers that want to verify that a widget is present in the tree should use this method.
```

?

Reply:
I'll add this.